lib/view_component/base.rb

```diff
@@ -173,8 +173,8 @@ class << self
       attr_accessor :source_location, :virtual_path
 
       # Render a component collection.
-      def with_collection(collection, **args)
-        Collection.new(self, collection, **args)
+      def with_collection(collection, collection_spacer: nil, **args)
+        Collection.new(self, collection, collection_spacer, **args)
```

Comment:
In keeping with our desire to match Rails APIs, how about we call this `spacer_component`? See https://guides.rubyonrails.org/layouts_and_rendering.html#spacer-templates.

```suggestion
      def with_collection(collection, spacer_component: nil, **args)
        Collection.new(self, collection, spacer_component, **args)
```